From 3e711c8ad1a45e87765e35a8026b414f29d679bc Mon Sep 17 00:00:00 2001
From: Greg Ungerer <gregory.ungerer@digi.com>
Date: Wed, 31 Oct 2018 13:57:44 +1000
Subject: [PATCH] user/net-snmp: support configuring on ARM64 machines

Add configure support for building on ARM64 machines. The code is fine,
we just need the configure front end support in place.
---
 config.guess | 7 +++++++
 config.sub   | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/config.guess b/config.guess
index ca2a03ca4f..5253cfd1d2 100755
--- a/config.guess
+++ b/config.guess
@@ -921,6 +921,13 @@ EOF
     ppc64:Linux:*:*)
 	echo powerpc64-unknown-linux-gnu
 	exit ;;
+    aarch64:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
+    aarch64_be:Linux:*:*)
+	UNAME_MACHINE=aarch64_be
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
 	  EV5)   UNAME_MACHINE=alphaev5 ;;
diff --git a/config.sub b/config.sub
index 2f24593c6a..c36ffbb8a7 100755
--- a/config.sub
+++ b/config.sub
@@ -238,6 +238,7 @@ case $basic_machine in
 	# Some are omitted here because they have special meanings below.
 	1750a | 580 \
 	| a29k \
+	| aarch64 | aarch64_be \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
 	| am33_2.0 \
@@ -319,6 +320,7 @@ case $basic_machine in
 	# Recognize the basic CPU types with company name.
 	580-* \
 	| a29k-* \
+	| aarch64-* | aarch64_be-* \
 	| alpha-* | alphaev[4-8]-* | alphaev56-* | alphaev6[78]-* \
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
-- 
2.17.1

